mlflow/gateway/gateway_app.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+from fastapi import FastAPI, Request, HTTPException
+import os
+from traceback import format_exc
+from typing import List
+import uvicorn
+
+from mlflow.gateway.constants import GATEWAY_VERSION, CONF_PATH_ENV_VAR
+from mlflow.gateway.handlers import Route, RouteConfig, _route_config_to_route, _load_gateway_config
+from mlflow.gateway.utils import _parse_url_path_for_base_url
+
+
+# Configured and initialized Gateway Routes state index
+ACTIVE_ROUTES: List[Route] = []
+
+app = FastAPI(
+    title="MLflow Model Gateway API",
+    description="The core gateway API for reverse proxy interface using remote inference "
+    "endpoints within MLflow",
+    version=GATEWAY_VERSION,
+)
+
+
+@app.get("/health")
+async def health():
+    return {"status": "OK"}
+
+
+@app.get("/gateway_url")
+async def gateway_url(request: Request):
+    return {"url": _parse_url_path_for_base_url(str(request.url))}
+
+
+@app.get("/get_route/{route:path}")
+async def get_route(route: str):
+    filtered = [x for x in ACTIVE_ROUTES if x.name == route]
+    if not filtered:
+        raise HTTPException(
+            status_code=404,
+            detail=f"The route '{route}' is not present or active on the server. Please "
+            "verify the route name.",
+        )
+    return {"route": filtered[0]}
+
+
+@app.get("/search_routes")
+async def search_routes(filter_condition: str):
+    # placeholder route listing functionality
+    routes = [route.json() for route in ACTIVE_ROUTES]
+    return {"routes": routes}
+
+
+@app.get("/list_all_routes")
+async def list_all_routes():
+    return [
+        {
+            "path": route.path,
+            "name": route.name,
+            "methods": sorted(route.methods),
+        }
+        for route in app.routes
+    ]
```

Comment:
I think we can remove this in favor of `search_routes`

Reply:
Question on this: search_routes will only return the dynamic configured routes, while this will return all routes present on the server ("/docs", "/health", "/openapi_x", etc; basically all of the quality of life things that fastapi brings). Do we want to expose an endpoint to list these or just leave it to the docs to explain what is available?

Reply:
(I did remove this endpoint in the latest push)